### PR TITLE
Join merge bug fix (proposed solution for #1843)

### PR DESCRIPTION
**Pull request by yphillip**
This PR is a proposed solution for the bug described in Issue #1843.

The `join-merge.vd` test is updated so that the `join-merge-1` sheet comes before the `join-merge-2` sheet, and the bug was then able to be reproduced. Before, it was the other way around, which did not accurately reflect the original use case of having the `join-merge-1` sheet be the initial original sheet.

Bug fix was to used `reversed` in the `calcValue()` function of the `MergeColumn` class so that new values are gotten from the last sheet(s) first.

- [ ] If contributing a core loader, [the loader checklist](https://visidata.org/docs/contributing#loader) was referenced.
- [ ] If registering an external plugin, [the plugin checklist](https://visidata.org/docs/contributing#plugins) was referenced.

### CODE REVIEW

**Comment by yphillip** [comment]
I manually tested the data set provided in #1843 and got the expected result (Bob has age of 35 in the merged sheet), but did not explicitly add it as a new test because I figured I could reproduce the bug in the existing `join-merge.vd` test. But if you do want me to explicitly add the new test, please let me know.

**Comment by saulpw** [comment]
Thanks, @yphillip!  Yes, if the existing test has been changed to elicit the bug before the change, then we don't need another test.  This all looks good to me, but I'll let @anjakefala weigh in before we merge it.

**Review by saulpw — APPROVED** [review]

**Comment by anjakefala** [comment]
Python 3.7 tests are failing because reversed dict views only became implemented in Python 3.8 :sob: https://github.com/python/cpython/pull/6827

If we want to continue to support 3.7, what is the recommended workaround? To listify it first?

**Comment by yphillip** [comment]
> Python 3.7 tests are failing because reversed dict views only became implemented in Python 3.8 😭 [python/cpython#6827](https://github.com/python/cpython/pull/6827)
> 
> If we want to continue to support 3.7, what is the recommended workaround? To listify it first?

That seems like the most straightforward solution to me too.

**Comment by anjakefala** [comment]
Thanks @yphillip! =)

**Comment by saulpw** [comment]
Thanks again for this, @yphillip!  If you'd like some VisiData stickers, please send me an email with your mailing address and I'd be happy to send you some.

**Pull request merged**
